### PR TITLE
KTOR-3302 | improve configuration of metric mame for http requests

### PR DESCRIPTION
**Pull request by mateusz-nalepa**
**Subsystem**
Client/Server, related modules

**Motivation**
[Add possibility to fully configure metricName](https://youtrack.jetbrains.com/issue/KTOR-3302)

**Solution**
Description of solution is present in issue.

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/src/io/ktor/server/metrics/micrometer/MicrometerMetrics.kt [hunk 1]
```diff
@@ -169,8 +172,18 @@ public class MicrometerMetrics private constructor(
      */
     public companion object Plugin : ApplicationPlugin<Application, Configuration, MicrometerMetrics> {
         private const val defaultBaseName: String = "ktor.http.server"
+        private const val defaultMetricName: String = "requests"
 
         private lateinit var baseName: String
+        private var metricName: String? = null
+
+        private fun resolveMetricName(): String {
```
[the remark this answers is not in the document]
Comment (mateuszNalepaAllegro) [reply ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/src/io/ktor/server/metrics/micrometer/MicrometerMetrics.kt]: Hello @rsinukov ,
I've deprecated `baseName` as you suggested. 

When we can expect version `2.0` of ktor? I would be so happy, to remove whole deprecated code from `MicrometerMetrics` 😄

**Review by rsinukov — APPROVED** [review]
LGTM!
2.0.0 will be released later this year, but we can not delete the deprecated API in it. We promised to have a deprecation cycle of at least one major release, so the earliest release to clean up code will be 3.0.0

**Comment by rsinukov** [comment]
@andreyaksenov please check api doc and deprecation message.

**Review by andreyaksenov — APPROVED** [review]

**Comment by rsinukov** [comment]
@mateusz-nalepa I see that there is a compile warning on 
```
 w: /mnt/agent/work/8d547b974a7be21f/ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/src/io/ktor/server/metrics/micrometer/MicrometerMetrics.kt: (227, 31): 'baseName: String' is deprecated. baseName is deprecated
  w: /mnt/agent/work/8d547b974a7be21f/ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/src/io/ktor/server/metrics/micrometer/MicrometerMetrics.kt: (233, 38): 'baseName: String' is deprecated. baseName is deprecated
```
Can you suppress deprecations, please?

**Comment by mateuszNalepaAllegro** [comment]
> @mateusz-nalepa I see that there is a compile warning on
> 
> ```
>  w: /mnt/agent/work/8d547b974a7be21f/ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/src/io/ktor/server/metrics/micrometer/MicrometerMetrics.kt: (227, 31): 'baseName: String' is deprecated. baseName is deprecated
>   w: /mnt/agent/work/8d547b974a7be21f/ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/src/io/ktor/server/metrics/micrometer/MicrometerMetrics.kt: (233, 38): 'baseName: String' is deprecated. baseName is deprecated
> ```
> 
> Can you suppress deprecations, please?

done

**Pull request merged**